### PR TITLE
Codebase refactor/rewrite

### PR DESCRIPTION
**Pull request by swedgwood**
Refactored/rewrote to:
 - Adhere to PEP8 guidelines
 - Adhere to a more procedural programming style
 - Include type-hinting
 - Improve readability

11/10 very important PR

### CODE REVIEW

File: hangman/__main__.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        guess = input('What is your guess? ').lower()
+
+        clear_screen()
+
+        if not len(guess) == 1:
+            print('One at a time please!')
+        elif guess not in letters:
+            print('Letters only please!')
+        elif guess in guessed_chars:
+            print("You've already tried that!")
```
Comment (thebeanogamer) [inline hangman/__main__.py:89]: Please don't mix quotes

File: hangman/__main__.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+def start_round(word: str):
+    round_finished = False
+    guessed_chars = ''
+    hanged_stage = 0
+
+    while not round_finished:
+        guess_blank = get_blanked_word(word, guessed_chars)
+
+        print("You've tried: " + guessed_chars)
```
Comment (thebeanogamer) [inline hangman/__main__.py:77]: Same quotes plz

**Review by thebeanogamer — CHANGES_REQUESTED** [review]

File: hangman/__main__.py (hunk 1 above)
Comment (Ge0rg3) [reply hangman/__main__.py:89]: https://www.python.org/dev/peps/pep-0008/#string-quotes

File: hangman/__main__.py (hunk 2 above)
Comment (Ge0rg3) [reply hangman/__main__.py:77]: https://www.python.org/dev/peps/pep-0008/#string-quotes

**Review by Ge0rg3 — APPROVED** [review]

File: hangman/__main__.py (hunk 2 above)
Comment (thebeanogamer) [reply hangman/__main__.py:77]: But mah uniformity!

**Review by thebeanogamer — APPROVED** [review]

**Pull request merged**